fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ConcreteAvailability.java

```diff
@@ -41,13 +44,35 @@ public ConcreteAvailability(
             @NotNull DataSourceMetadataService metadataService
     ) {
         this.name = tableName;
-        this.columns = columns;
         this.metadataService = metadataService;
+        this.names = Collections.singleton(tableName);
+        this.columns = columns;
+        this.columnNames = columns.stream()
+                .map(Column::getName)
+                .collect(Collectors.toSet());
```

Comment:
Should also be made into an `ImmutableSet`, though it matters a _little_ less, since it's never returned directly. Better to be safe though.